src/definitions/x509/validation/extensions/crl_distribution_points.rs

```diff
@@ -0,0 +1,177 @@
+use const_oid::AssociatedOid;
+use der::Decode;
+use x509_cert::ext::{
+    pkix::{
+        name::{DistributionPointName, GeneralName},
+        CrlDistributionPoints,
+    },
+    Extension,
+};
+
+use super::{Error, ExtensionValidator};
+
+/// CRLDistributionPoints validation for all certificate profiles.
+pub struct CrlDistributionPointsValidator;
+
+impl CrlDistributionPointsValidator {
+    fn check(crl_distribution_points: CrlDistributionPoints) -> Vec<Error> {
+        if crl_distribution_points.0.is_empty() {
+            return vec!["expected one or more distribution points".into()];
+        }
+        let mut errors = vec![];
+        for point in crl_distribution_points.0.into_iter() {
+            if point.crl_issuer.is_some() {
+                errors.push(format!(
+                    "crl_issuer cannot be set, but is set for: {point:?}"
+                ))
+            }
+
+            if point.reasons.is_some() {
+                errors.push(format!("reasons cannot be set, but is set for: {point:?}",))
+            }
+
+            if !point
+                .distribution_point
+                .as_ref()
+                .is_some_and(|dpn| match dpn {
+                    DistributionPointName::FullName(names) => names
+                        .iter()
+                        .any(|gn| matches!(gn, GeneralName::UniformResourceIdentifier(_))),
+                    DistributionPointName::NameRelativeToCRLIssuer(_) => false,
+                })
+            {
+                errors.push(format!("point is invalid: {point:?}",))
+            }
+        }
+        errors
+    }
+}
+
+impl ExtensionValidator for CrlDistributionPointsValidator {
+    fn oid(&self) -> const_oid::ObjectIdentifier {
+        CrlDistributionPoints::OID
+    }
+
+    fn ext_name(&self) -> &'static str {
+        "CrlDistributionPoints"
+    }
+
+    fn validate(&self, extension: &Extension) -> Vec<Error> {
```

Comment:
I'm not 100% that I'm fully following the entire flow, but I think you should still check here that the CrlDistributionPoint extension is marked as non-critical.

Reply:
My understanding is that the `critical` flag indicates that the consumer MUST understand the extension, otherwise it should fail the transaction. In these cases, the 18013-5 spec says that these extensions shall not be critical. But from a practical perspective we _do_ understand these extensions, so I don't think it matters whether or not the extension is actually marked as critical. I'm leaning more toward a slightly more permissive implementation in cases where security isn't impacted, rather than strictly enforcing every requirement.

Reply:
I see what you mean. My point of view is that since this is open source, developers may use this to validate their own certificates' iso compliance and this still leaves some room for them to do it incorrectly. Not a hill I'd die on though.

Reply:
Makes sense. We probably want a "strict" mode and "permissive" mode to allow for that as we've discussed before. I think that's out of the scope of this work, but I'll make sure we have a task that captures this.